src/test/ui/consts/const-float-bits-conv.rs

```diff
@@ -0,0 +1,75 @@
+// run-pass
+
+#![feature(const_panic)]
+#![feature(const_if_match)]
+#![feature(const_float_bits_conv)]
+
+macro_rules! const_assert {
+    ($a:expr) => {
+        {
+            const _: () = assert!($a);
+            assert!($a);
+        }
+    };
+    ($a:expr, $b:expr) => {
+        {
+            const _: () = assert!($a == $b);
+            assert_eq!($a, $b);
```

Comment:
Maybe add something like `-C opt-level=0 -Zmir-opt-level=0` to make sure we don't optimize the run-time code? And add some curly braces to also defeat promotion? (AFAIK `&{x}` does not get promoted but I might misremember.)